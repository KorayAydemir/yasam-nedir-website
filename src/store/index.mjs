import { configureStore } from '@reduxjs/toolkit';
import { rootReducer } from './reducers/index.mjs'; 
import { rtkSanityApi } from '../client/fetcher.mjs'
    
export const setupStore = (preloadedState, composeEnhancers) => {
    return configureStore({
        reducer: rootReducer,
        preloadedState,
        middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(rtkSanityApi.middleware),
    });
}
