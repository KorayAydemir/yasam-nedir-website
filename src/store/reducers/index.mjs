import yasamBilim from '../../app/domain/yasambilim/reducer/index.mjs';
import layout from '../../app/domain/layout/reducer/index.mjs';
import landing from '../../app/domain/landing/reducer/index.mjs';
import { rtkSanityApi } from '../../client/fetcher.mjs';

export const rootReducer = {
    yasamBilim,
    layout,
    landing,
    [rtkSanityApi.reducerPath]: rtkSanityApi.reducer
}
