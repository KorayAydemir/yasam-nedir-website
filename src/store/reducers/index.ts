import yasamBilim from '../../app/domain/yasambilim/reducer/index';
import layout from '../../app/domain/layout/reducer/index';
import landing from '../../app/domain/landing/reducer/index';
import { rtkSanityApi } from '../../fetch/index';

export const rootReducer = {
    yasamBilim,
    layout,
    landing,
    [rtkSanityApi.reducerPath]: rtkSanityApi.reducer
}
