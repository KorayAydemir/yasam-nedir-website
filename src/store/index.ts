import { configureStore } from '@reduxjs/toolkit';
import { rootReducer } from './reducers/index'; 
import { rtkSanityApi } from '../fetch/index'
    
export const setupStore = (preloadedState, composeEnhancers) => {
    return configureStore({
        reducer: rootReducer,
        preloadedState,
        middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(rtkSanityApi.middleware),
    });
}
