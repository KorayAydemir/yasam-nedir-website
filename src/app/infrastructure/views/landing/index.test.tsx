import renderer from "react-test-renderer";
import Landing from "./index";
import { Provider } from "react-redux";
import { setupStore } from "../../../../store";
import { BrowserRouter } from "react-router-dom";

describe("Views > Landing", () => {
    it("should render", () => {
        const store = setupStore({});
        const comp = renderer.create(
            <BrowserRouter>
                <Provider store={store}>
                    <Landing />
                </Provider>
            </BrowserRouter>
        );

        const tree = comp.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
