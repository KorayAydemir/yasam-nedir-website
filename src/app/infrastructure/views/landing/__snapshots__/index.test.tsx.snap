// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Views > Landing should render 1`] = `
[
  <h1
    className="underline"
  >
    Hello from Landing ðŸ‘½
  </h1>,
  <a
    href="/yasambilim"
    onClick={[Function]}
  >
    GOTO yasambilim
  </a>,
]
`;
